yt_dlp/extractor/piramidetv.py

```diff
@@ -34,27 +34,28 @@ class PiramideTVIE(InfoExtractor):
     }]
 
     def _extract_video(self, video_id):
-        video_data = self._download_json(
-            f'https://hermes.piramide.tv/video/data/{video_id}', video_id, fatal=False)
-        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
-            f'https://cdn.piramide.tv/video/{video_id}/manifest.m3u8', video_id, fatal=False)
-        next_video = traverse_obj(video_data, ('video', 'next_video', 'id', {str}))
-        return next_video, {
-            'id': video_id,
-            'formats': formats,
-            'subtitles': subtitles,
-            'webpage_url': f'https://piramide.tv/video/{video_id}',
-            'original_url': f'https://piramide.tv/video/{video_id}',
-            **traverse_obj(video_data, ('video', {
-                'id': ('id', {str}),
-                'title': ('title', {str}),
-                'description': ('description', {str}),
-                'thumbnail': ('media', 'thumbnail', {url_or_none}),
-                'channel': ('channel', 'name', {str}),
-                'channel_id': ('channel', 'id', {str}),
-                'timestamp': ('date', {parse_iso8601}),
-            })),
-        }
+        if video_data := self._download_json(
+                f'https://hermes.piramide.tv/video/data/{video_id}', video_id, fatal=False):
```

Comment:
Because fatal=False, extract_m3u8_formats_and_subtitles will wait until timeout when the video id is actually not exist.
This change is made to skip the process if video_data is not available.

Reply:
Why should video data extraction be mandatory? If they change the metadata endpoint, but the m3u8 is still in the same place, wouldn't it be better if the extractor could still download something?

If you want to check for missing formats, you could add:
```suggestion
        if not formats:
            self.raise_no_formats('No video formats found')
```
But even that is unnecessary imo.
Core code will throw an error if an empty formats array is returned.

Reply:
I'd do an early exit and avoid the indent:
```py
video_data = self._download_json(...)
if not video_data:
    return None, {'id': video_id}
formats, subtitles = self._extract_m3u8_formats_and_subtitles(...)
```